File: apps/chat-ios/AmazonChimeSDKMessagingDemo/entity/Message.swift
//
//  Message.swift
//  AmazonChimeSDKMessagingDemo
//
//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
//  SPDX-License-Identifier: MIT-0
//

import Foundation

@objcMembers public class Message: NSObject, Codable {
    let headers: [String: String]
    let payload: String?

    enum CodingKeys: String, CodingKey {
        case headers = "Headers"
        case payload = "Payload"
    }
}

File: apps/chat-android/messaging-websocket/src/main/java/com/amazonaws/services/chime/sdk/messaging/Message.kt
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

package com.amazonaws.services.chime.sdk.messaging

data class Message(
    val Headers: Map<String, String>,
    val Payload: String
)
